src/transcript-plugin.tsx

```diff
@@ -1,401 +1,294 @@
-} from "./utils";
-import { DownloadPrintMenu } from "./components/download-print-menu";
-
-const pluginName = `playkit-js-transcript`;
-
-const logger = getContribLogger({
-  class: "TranscriptPlugin",
-  module: "transcript-plugin"
-});
-
-const { get } = ObjectUtils;
-
-interface TranscriptPluginConfig {
-  expandOnFirstPlay: boolean;
-  showTime: boolean;
-  position: KitchenSinkPositions;
-  scrollOffset: number; // distance between top border of transcript container and active caption on auto-scroll
-  searchDebounceTimeout: number; // debounce on search
-  searchNextPrevDebounceTimeout: number; // debounce on jump between prev/next search result
-  downloadDisabled: boolean; // disable download menu
-  printDisabled: boolean; // disable print menu
-  expandMode: KitchenSinkExpandModes; // over or pushing the player
+import {h} from 'preact';
+import {ObjectUtils} from './utils';
+import {ui} from 'kaltura-player-js';
+import {KalturaCaptionAsset} from 'kaltura-typescript-client/api/types/KalturaCaptionAsset';
+import {PluginButton} from './components/plugin-button/plugin-button';
+import {Transcript} from './components/transcript';
+import {getConfigValue, isBoolean, makePlainText, itemTypesOrder, prepareCuePoint} from './utils';
+import {TranscriptConfig, PluginPositions, PluginStates, HighlightedMap, CaptionItem, ItemData, ItemTypes, CuePoint} from './types';
+import {DownloadPrintMenu, downloadContent, printContent} from './components/download-print-menu';
+
+const {SidePanelModes, SidePanelPositions, ReservedPresetNames, ReservedPresetAreas} = ui;
+const {get} = ObjectUtils;
+const {Tooltip} = KalturaPlayer.ui.components;
+
+interface TimedMetadataEvent {
+  payload: {
+    cues: Array<CuePoint>;
+  };
 }
 
-export class TranscriptPlugin implements OnMediaLoad, OnPluginSetup, OnMediaUnload {
-  private _kitchenSinkItem: KitchenSinkItem | null = null;
-  private _upperBarItem: UpperBarItem | null = null;
+export class TranscriptPlugin extends KalturaPlayer.core.BasePlugin {
+  static defaultConfig: TranscriptConfig = {
+    expandMode: SidePanelModes.ALONGSIDE,
+    expandOnFirstPlay: true,
+    showTime: true,
+    position: SidePanelPositions.BOTTOM,
+    scrollOffset: 0,
+    searchDebounceTimeout: 250,
+    searchNextPrevDebounceTimeout: 100,
+    downloadDisabled: false,
+    printDisabled: false
+  };
+  private _activeCaptionMapId: string = '';
+  private _activeCuePointsMap: HighlightedMap = {};
+  private _highlightedGroup: Array<ItemData> = [];
+  private _captionMap: Map<string, Array<ItemData>> = new Map();
   private _isLoading = false;
   private _hasError = false;
-  private _entryId = "";
   private _captionsList: KalturaCaptionAsset[] = []; // list of captions
   private _captions: CaptionItem[] = []; // parsed captions
-  private _kalturaClient = new KalturaClient();
-  private _transcriptLanguage = "default";
-  private _triggeredByKeyboard  = false;
+  private _transcriptLanguage = 'default';
```

Comment:
`_transcriptLanguage` uses only for creation file-name (download captions).
Better - take active language from `this.player.getTracks(this.player.Track.TEXT)`